servers/src/test/java/tachyon/master/next/filesystem/meta/InodeDirectoryTests.java

```diff
@@ -0,0 +1,185 @@
+  }
+
+  @Test
+  public void reverseIdTest() {
+    InodeDirectory inode1 = createInodeDirectory();
+    inode1.reverseId();
+    Assert.assertEquals(-1, inode1.getId());
+  }
+
+  @Test
+  public void sameIdChildrenTest() {
+    InodeDirectory inodeDirectory = createInodeDirectory();
+    InodeFile inodeFile1 = createInodeFile(1);
+    inodeDirectory.addChild(inodeFile1);
+    inodeDirectory.addChild(inodeFile1);
+    Assert.assertTrue(inodeDirectory.getChildrenIds().get(0) == createBlockId(1));
+    Assert.assertEquals(1, inodeDirectory.getNumberOfChildren());
+  }
+
+  @Test
+  public void setLastModificationTimeTest() {
+    long createTimeMs = System.currentTimeMillis();
+    long modificationTimeMs = createTimeMs + 1000;
+    InodeDirectory inodeDirectory = createInodeDirectory();
+    Assert.assertEquals(createTimeMs, inodeDirectory.getLastModificationTimeMs());
+    inodeDirectory.setLastModificationTimeMs(modificationTimeMs);
+    Assert.assertEquals(modificationTimeMs, inodeDirectory.getLastModificationTimeMs());
+  }
+
+  @Test
+  public void setNameTest() {
+    InodeDirectory inode1 = createInodeDirectory();
+    Assert.assertEquals("test1", inode1.getName());
+    inode1.setName("test2");
+    Assert.assertEquals("test2", inode1.getName());
+  }
+
+  @Test
+  public void setParentIdTest() {
+    InodeDirectory inode1 = createInodeDirectory();
+    Assert.assertEquals(0, inode1.getParentId());
+    inode1.setParentId(2);
+    Assert.assertEquals(2, inode1.getParentId());
+  }
+
+  @Test
+  public void getChildTest() {
+    // large number of small files
+    InodeDirectory inodeDirectory = createInodeDirectory();
+    int nFiles = (int) 1E5;
+    Inode[] inodes = new Inode[nFiles];
+    for (int i = 0; i < nFiles; i ++) {
+      inodes[i] = createInodeFile(i + 1);
+      inodeDirectory.addChild(inodes[i]);
+    }
+
+    Runtime runtime = Runtime.getRuntime();
+    LOG.info(String.format("Used Memory = %dB when number of files = %d",
+        runtime.totalMemory() - runtime.freeMemory(), nFiles));
+
+    long start = System.currentTimeMillis();
+    for (int i = 0; i < nFiles; i ++) {
+      Assert.assertEquals(inodes[i], inodeDirectory.getChild(createBlockId(i + 1)));
+    }
+    LOG.info(String.format("getChild(int fid) called sequentially %d times, cost %d ms", nFiles,
+        System.currentTimeMillis() - start));
+
+    start = System.currentTimeMillis();
+    for (int i = 0; i < nFiles; i ++) {
+      Assert.assertEquals(inodes[i], inodeDirectory.getChild(String.format("testFile%d", i + 1)));
+    }
+    LOG.info(String.format("getChild(String name) called sequentially %d times, cost %d ms", nFiles,
+        System.currentTimeMillis() - start));
+  }
+
+  private long createBlockId(long containerId) {
+    return BlockId.createBlockId(containerId, BlockId.getMaxSequenceNumber());
+  }
+
+  private static InodeDirectory createInodeDirectory() {
+    return new InodeDirectory("test1", 1, 0, System.currentTimeMillis());
```

Comment:
shall we do similar thing as `createInodeFile` that to take an id and construct the directory name based on id?
Because in some test cases, we are assserting test1 and test2, that will simply the code.